File: ios/ScrollEdgeEffectViewManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE (ScrollEdgeEffectViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(scrollViewTag, NSNumber)
RCT_EXPORT_VIEW_PROPERTY(edge, NSString)
RCT_EXPORT_VIEW_PROPERTY(effectStyle, NSString)

@end

File: ios/ScrollEdgeEffectViewManager.swift
import React
import UIKit

@objc(ScrollEdgeEffectViewManager)
final class ScrollEdgeEffectViewManager: RCTViewManager {
  override func view() -> UIView! {
    let view = ScrollEdgeEffectView()
    view.bridge = bridge

    return view
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
